internal/pkg/instrumentation/manager.go

```diff
@@ -398,11 +419,189 @@ func (m *Manager) loadProbes() error {
 	return nil
 }
 
+func (m *Manager) LoadProbe(i ProbeReference, name probe.ID, cfg Config) error {
+	m.logger.Info("loading probe", "name", name)
+
+	spec, err := i.probe.Spec()
+	if err != nil {
+		return errors.Join(err, m.cleanup())
+	}
+
+	err = m.InjectProbeConsts(i.probe, spec)
+	if err != nil {
+		return errors.Join(err, m.cleanup())
+	}
+
+	c, err := utils.InitializeEBPFCollection(spec, m.collectionOpts)
+	if err != nil {
+		return errors.Join(err, m.cleanup())
+	}
+	i.collection = c
+
+	reader, err := i.probe.InitStartupConfig(c, cfg.SamplingConfig)
+	if err != nil {
+		return errors.Join(err, m.cleanup())
+	}
+	i.closers = append(i.closers, reader)
+
+	return nil
+}
+
+func (m *Manager) InjectProbeConsts(i probe.Probe, spec *ebpf.CollectionSpec) error {
```

Comment:
Should this be exported? It should only ever be called by the manager and never external to this package, right?